lib/genesis/test_utils.go

```diff
@@ -100,23 +102,25 @@ func CreateTestGenesisJSONFile(t *testing.T, asRaw bool) (filename string) {
 
 // NewTestGenesisWithTrieAndHeader generates genesis, genesis trie and genesis header
 func NewTestGenesisWithTrieAndHeader(t *testing.T) (*Genesis, *trie.Trie, *types.Header) {
-	gen, err := NewGenesisFromJSONRaw("../../chain/gssmr/genesis.json")
-	if err != nil {
-		gen, err = NewGenesisFromJSONRaw("../../../chain/gssmr/genesis.json")
-		require.NoError(t, err)
-	}
+	_, fullpath, _, _ := runtime.Caller(0)
+	rootDir := path.Dir(path.Dir(path.Dir(fullpath))) // same as ../../..
+	genesisPath := path.Join(rootDir, "chain/gssmr/genesis.json")
+
+	gen, err := NewGenesisFromJSONRaw(genesisPath)
+	require.NoError(t, err)
 
 	tr, h := newGenesisTrieAndHeader(t, gen)
 	return gen, tr, h
 }
 
 // NewDevGenesisWithTrieAndHeader generates test dev genesis, genesis trie and genesis header
 func NewDevGenesisWithTrieAndHeader(t *testing.T) (*Genesis, *trie.Trie, *types.Header) {
-	gen, err := NewGenesisFromJSONRaw("../../chain/dev/genesis.json")
-	if err != nil {
-		gen, err = NewGenesisFromJSONRaw("../../../chain/dev/genesis.json")
-		require.NoError(t, err)
-	}
+	_, fullpath, _, _ := runtime.Caller(0)
+	rootDir := path.Dir(path.Dir(path.Dir(fullpath))) // same as ../../..
+	genesisPath := path.Join(rootDir, "chain/dev/genesis.json")
```

Comment:
This is risky since depending where the caller is, this won't work. Maybe implementing a search function would be nice (it's also needed when we want to move directories to `internal/` later). What I'm thinking of is to:

1. Search for directory containing `go.mod` with `module github.com/ChainSafe/gossamer` by going up everytime it's not found. That way we can find the root of the repo.
2. Search for the chain directory by walking from the root, and then use that path together with the root to have an absolute path to the desired JSON file.

This can be shared in the several exported helpers test functions in this file.

Reply:
`runtime.Caller(0)` with argument 0 returns the caller of `runtime.Caller` function so it will always be the current folder where the function is, however I agreed that it can be improved to be more flexibe, because right now we need to exec 3 hard coded jumps to achieve the root, I will improve given your suggestion!